File: examples/producer/typescript/src/producer/responses/users.request.ts
import { Response } from "../implementation/responses/users.request";
import {
	Context,
	Producer,
	Identification,
	Filter,
	Protocol,
} from "../implementation/responses";
import { Scope } from "../implementation/scope";

export function response(
	request: Protocol.Users.Request,
	scope: Scope
): Promise<Response> {
	return Promise.resolve(
		new Response(
			new Protocol.Users.Response({
				users: scope.context.getUsers(),
			})
		)
	);
}

File: examples/producer/rust/src/producer/responses/users_request.rs
use super::{identification, producer::Control, protocol, scope::Scope, Context};
use clibri::server;

#[allow(unused_variables)]
pub async fn response<E: server::Error, C: server::Control<E>>(
    request: &protocol::Users::Request,
    scope: &mut Scope<'_, E, C>,
) -> Result<protocol::Users::Response, protocol::Users::Err> {
    Ok(protocol::Users::Response {
        users: scope.context.get_users(),
    })
}
